src/graph/validator/MatchValidator.cpp

```diff
@@ -126,6 +126,9 @@ Status MatchValidator::validateImpl() {
 // \param matchClauseCtx the context of current match clause
 Status MatchValidator::validatePath(const MatchPath *path, MatchClauseContext &matchClauseCtx) {
   matchClauseCtx.paths.emplace_back();
+  if (path->alias() == nullptr) {
+    const_cast<MatchPath *>(path)->setPredicate();
+  }
```

Comment:
What does this mean, that paths without aliases are all predicates?

Reply:
yes

Reply:
prefer some comments about this update

Reply:
done

Reply:
`setPredicate()` this function is designed to mark a path pattern to the predicate not a path pattern without aliases.
I cannot get your point.

Reply:
Yes, but I saw the use of the predicate function when generating an execution plan, which means the same as not generating a path structure

Reply:
Annotations have been added where used

Reply:
> Yes, but I saw the use of the predicate function when generating an execution plan, which means the same as not generating a path structure

It is true that a pattern predicate does not produce a path data at runtime, but that does not mean something that doesn't produce a path data should be set as a path predicate.

Reply:
what is your suggestion? Redefine a variable?